cmd/out/main.go

```diff
@@ -127,8 +127,9 @@ func main() {
 		SkipSSLValidation: input.Source.SkipSSLValidation,
 	})
 
+	getPrefixFunction := uploader.NewPrefixFinder(client, input.Source.ProductSlug)
 	uploaderClient := uploader.NewClient(uploader.Config{
-		FilepathPrefix: input.Params.FilepathPrefix,
+		GetFilePrefix: 	getPrefixFunction,
```

Comment:
Why is this still in the config? I thought FilepathPrefix is being removed. This is a little confusing. 

How about just calling this boxed closure directly at https://github.com/pivotal-cf/pivnet-resource/blob/release-improvements/uploader/uploader.go#L63 ?

Another idea, instead of returning anonymous functions that close over data, consider using a struct. It will create a new type which is easier to maintain than anonymous functions. Also, when i did a search for anonymous functions in our codebase, i only found this new reference. This means that this is a new pattern that is being introduced. I think this pattern is not needed. 

Thoughts?